Running netcheck results in a runtime error: invalid memory address or nil pointer dereference

### What is the issue?

When running tailscale netcheck in cmd, the following error occurs:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x93c5b3]

goroutine 50 [running]:
tailscale.com/net/netcheck.(*reportState).runProbe(0xc00027c000, {0xbc9690, 0xc000273810}, 0x0?, {0x0, {0xc00020a570, 0x2}, 0x1, 0x0}, 0xc000228420)
        tailscale.com@v1.1.1-0.20230422043621-e3b2250e2654/net/netcheck/netcheck.go:1569 +0x653
created by tailscale.com/net/netcheck.(*Client).GetReport.func6
        tailscale.com@v1.1.1-0.20230422043621-e3b2250e2654/net/netcheck/netcheck.go:998 +0x76
```

### Steps to reproduce

Open cmd.exe, run the command ``tailscale netcheck``, it fails.

### Are there any recent changes that introduced the issue?

This is a fairly new install, I didn't make any changes after install as far as I recall.

### OS

Windows

### OS version

Win10 22H2

### Tailscale version

1.39.194

### Other software

I'm running NordVPN which might potentially be causing the issue. I added tailscaled.exe, tailscale.exe and the Tailscale GUI client to the split tunneling list to try and avoid issues.

### Bug report

BUG-4fab5752eefc5f1b4358829341a442f10e86e1519ee9cca7f78d398eb07048ca-20230422212303Z-48106b53f4cbf11b

--- a/net/netcheck/netcheck.go
+++ b/net/netcheck/netcheck.go
@@ -40,6 +40,7 @@ import (
 	"tailscale.com/types/logger"
 	"tailscale.com/types/nettype"
 	"tailscale.com/types/opt"
+	"tailscale.com/types/ptr"
 	"tailscale.com/util/clientmetric"
 	"tailscale.com/util/mak"
 )
@@ -942,6 +943,16 @@ func (c *Client) GetReport(ctx context.Context, dm *tailcfg.DERPMap) (_ *Report,
 				rs.pc6 = u6
 				go c.readPackets(ctx, u6)
 			}
+		}
+
+		// If our interfaces.State suggested we have IPv6 support but then we
+		// failed to get an IPv6 sending socket (as in
+		// https://github.com/tailscale/tailscale/issues/7949), then change
+		// ifState.HaveV6 before we make a probe plan that involves sending IPv6
+		// packets and thus assuming rs.pc6 is non-nil.
+		if rs.pc6 == nil {
+			ifState = ptr.To(*ifState) // shallow clone
+			ifState.HaveV6 = false
 		}
 	}
 

